src/ShellClients/ShellClientsManager.vala

```diff
@@ -34,10 +34,50 @@ public class Gala.ShellClientsManager : Object {
     construct {
         notifications_client = new NotificationsClient (wm.get_display ());
 
-        var shell_settings = new Settings ("io.elementary.desktop.wm.shell");
-        var clients = shell_settings.get_value ("trusted-clients");
-        foreach (var client in clients) {
-            protocol_clients += new ManagedClient (wm.get_display (), client.get_strv ());
+        start_clients.begin ();
+    }
+
+    private async void start_clients () {
+        var file = File.new_build_filename (Config.SYSCONFDIR, "io.elementary.desktop.wm.shell");
```

Comment:
Here I think you want to get system config dirs from glib: https://valadoc.org/glib-2.0/GLib.Environment.get_system_config_dirs.html

here’s an example of looping over the list so we prioritize user overrides https://github.com/elementary/switchboard-plug-parental-controls/blob/49100787dcfd354f6a249c9f8307182d3e659226/src/daemon/Server.vala#L297

Reply:
Done :)